cmd/crossplane/core/core.go

```diff
@@ -136,6 +138,11 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 		return errors.Wrap(err, "Cannot get config")
 	}
 
+	cfg.WarningHandler = rest.NewWarningWriter(os.Stderr, rest.WarningWriterOptions{
```

Comment:
@hasheddan @turkenh what do you think of this approach I found from a Jordon Liggitt blog post? https://kubernetes.io/blog/2020/09/03/warnings/#customize-client-handling

It does indeed result in k8s REST API warnings only being printed once per warning, but I can't say I fully understand other potential impact of making this change, so it would be good to hear your thoughts! 🙇‍♂️

Reply:
Thanks for finding this @jbw976! I was initially wondering if we could turn off warnings entirely:

```
  // set the warning handler for this client to ignore warnings
  config.WarningHandler = rest.NoWarnings{}
 ```

My rationale here is that warnings emitted by core Crossplane are not necessarily something I would consider “user actionable”. Perhaps the exception to that rule is if a user is composing resources and is using a deprecated API in their `Composition`. A core Crossplane log _could_ encourage them to update their `Composition` to a newer version.

It looks like the only downside of using the deduplication strategy is increased memory usage, but it looks like memory usage should [scale linearly](https://github.com/kubernetes/client-go/blob/7c4f8c88f6cb8817c04724e28bc783f3fd1f4b00/rest/warnings.go#L117) with the number of _type_ of warnings, not the total number of warnings, so I’m not too concerned about that (unless we start experiencing some warning that has a dynamic message that includes a time stamp, counter, etc.).

Overall, it seems like deduplication is at least a good first step, and we can make the decision whether to turn off entirely in the future if needed.

Reply:
That all makes sense to me @hasheddan, thank you for diving into this a bit and then sharing your thoughts! 🙇‍♂️